datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -706,17 +670,10 @@ impl LogicalPlan {
                     }))
                 }
             },
-            LogicalPlan::Window(Window {
-                window_expr,
-                schema,
-                ..
-            }) => {
+            LogicalPlan::Window(Window { window_expr, .. }) => {
                 assert_eq!(window_expr.len(), expr.len());
-                Ok(LogicalPlan::Window(Window {
-                    input: Arc::new(inputs[0].clone()),
-                    window_expr: expr,
-                    schema: schema.clone(),
-                }))
```

Comment:
Since input/window_expr will change, we shouldn't reuse previous schema.

Reply:
maybe @mustafasrepo  remembers the context -- I vaguely remember something like this but can't remember the specifics

Reply:
> This looks like a potential bug. Since input/window_expr will change, we shouldn't reuse previous schema.

Aggree that, previous version wasn't robust to expression and input changes. This version is much better. Thanks @viirya for catching this.